### ISSUE TITLE
raftexample deletes nodes and adds old nodes abnormally

### ISSUE THREAD

**Issue by ws-weisb** [etcd-io/etcd#18777]
### Bug report criteria

- [X] This bug report is not security related, security issues should be disclosed privately via [etcd maintainers](mailto:etcd-maintainers@googlegroups.com).
- [X] This is not a support request or question, support requests or questions should be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
- [X] You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
- [X] Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.

### What happened?

raftexample

Q1: Deleting the data on the deleted machine is useless, will replay synced data after restart. All machine configurations need to be removed. Is this expected?

![image](https://github.com/user-attachments/assets/f7265f35-04ad-4a16-be45-fc95e74a8e7f)

relation issue https://github.com/etcd-io/etcd/issues/15380#issuecomment-2425704103_

steps:
1. run cluster by Procfile
2. curl DELETE /3 on ID 1
3. curl POST /3 on ID 1
4. rm -fr raftexample-3*
5. restart ID 3 by same command

Q2: Why https://github.com/etcd-io/etcd/pull/13645 is not included in v3.5.16
tag v3.5.16
![image](https://github.com/user-attachments/assets/031ca720-d937-4820-8df0-9a30d6a0480a)

### What did you expect to happen?

Q1: only delete the data of the previously removed nodes instead of all nodes
Q2: don't panic when remove unknown id, the previous commit should be in v3.5.16

### How can we reproduce it (as minimally and precisely as possible)?

Q1: 
run raftexample on v3.5.16
1. run cluster by Procfile
2. curl DELETE /3 on ID 1
3. curl POST /3 on ID 1
4. rm -fr raftexample-3*
5. restart ID 3 by same command

Q2:
run raftexample on v3.5.16
curl DELETE /4 on any node

### Etcd version (please run commands below)

<details>

git tag v3.5.16

```console
$ etcd --version
no

$ etcdctl version
no
```

</details>

### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

</details>

### Etcd debug information (please run commands below, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>

**Comment by ahrtr** [comment]
> Q1: Deleting the data on the deleted machine is useless, will replay synced data after restart. All machine configurations need to be removed. Is this expected?

It should be expected. A new member will sync data from other members automatically on startup. Note that we are not actively maintaining raftexample. Once https://github.com/etcd-io/raft/issues/2 is resolved, then we will remove it from etcd repo.



> Q2: Why #13645 is not included in v3.5.16

Note that issue https://github.com/etcd-io/etcd/issues/13466 could only be reproduced in 3.5.0. So no need to backport the fix to release-3.5. Read https://github.com/ahrtr/etcd-issues/tree/master/issues/13466

But It seems that you are still talking about raftexample.

**Comment by maxiloEmmmm** [comment]
@ahrtr That’s right. Q2 still refers to raftexample. Is this error still possible in 3.5.16 because raftexample is not maintained?
